tracknow/web/src/components/Post/Post.tsx

```diff
@@ -20,7 +17,24 @@ type PostProps = {
 // homepage posts( recent) posts of users
 export const HomePost: React.FC<PostProps> = ({ laptimes, fetchMoreData, hasMore }) => {
 
-    const { laptime_loading } = useLaptimes();
+    // intersection observation api instead of the bad infinite scroll component
```

Comment:
I resolved the bug by removing the InfiniteScroll component (an external library) from Post.tsx and replacing it with a custom Intersection Observer method to load more data. This approach is more efficient and reliable.

The InfiniteScroll component has a flaw: it requires additional pixels beyond the viewport height to trigger the loading of more data. This means that if you restrict the parent Flex component to 100vh, the component won't work correctly. For example:


`<Flex height='100vh'>
  <InfiniteScroll />
</Flex>`
In this scenario, the height must be greater than 100vh for the component to function properly. It's unclear how many extra pixels are needed to reach the 'end' of the page, making it impractical when you want to restrict the page height to 100vh and keep it in place.

By using a custom Intersection Observer method, I was able to achieve a more seamless and predictable infinite scrolling experience.